sdk/cosmosdb/cosmos/review/cosmos.api.md

```diff
@@ -88,6 +114,18 @@ export class ChangeFeedResponse<T> {
     readonly statusCode: number;
 }
 
+// @public
+export abstract class ChangeFeedStartFrom {
+    // Warning: (ae-forgotten-export) The symbol "ChangeFeedStartFromBeginning" needs to be exported by the entry point index.d.ts
+    static Beginning(cfResource?: PartitionKey | PartitionKeyRange | EpkRange): ChangeFeedStartFromBeginning;
```

Comment:
Yes, `PartitionKeyRange` is public in JS SDK.

Reply:
Interesting, out of curiosity, do you know why these are public? And should we just take PartitionKey and EpkRange like the other languages instead?

A CF pull with PartitionKeyRange as input is **not split proof** because you cannot honor the PartitionKeyRange if a split happens.

Reply:
I am not aware why this was made public. 

A CF pull with PartitionKeyRange as input is split proof as split is handled internally. Even if a user passes a PartitionKeyRange that has been split, the children feed ranges are added internally into the queue. 

I can make the changes to just accept PartitionKey or FeedRange to make it similar to other SDKs. Internally PartitionKeyRanges will be converted into FeedRanges(EpkRanges) and returned as an array to the user.

Reply:
> A CF pull with PartitionKeyRange as input is split proof as split is handled internally. Even if a user passes a PartitionKeyRange that has been split, the children feed ranges are added internally into the queue. Am I missing something here?

You grab the PKRangeId and convert it to EPK Range and handle future splits, yes, but what if the PKRangeId being passed in is already split/does not exist anymore? You cannot even convert it to EPK Range, right?

Reply:
PKRangeId present in PKRange is not directly used. minInclusive and maxExclusive present in PKRange are used to fetch overlapping ranges.

Reply:
I have made changes to just accept FeedRange and PartitionKey as input. A new `getFeedRanges` is added which returns feedranges to the user. Internally, this `getFeedRanges` convert the PKRanges into FeedRanges. 
FeedRanges is now converted into internal class and check is placed to avoid tampering with feed range returned.